### PR TITLE
Refactorings around predicates

### PR DESCRIPTION
**Pull request by Mats-SX**
The recent bugfixes to `Ands` and `Ors` classes exposed a few opportunities to clean up the code, and allow newer siblings of them to be inserted in the new inheritance pattern introduced by `CompositeBooleanPredicate` class.

### CODE REVIEW

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def isMatch(m: ExecutionContext)(implicit state: QueryState): Option[Boolean] = Ors(NonEmptyList(a, b)).isMatch(m)
+
+  override def toString: String = "(" + a + " OR " + b + ")"
+  def containsIsNull = a.containsIsNull||b.containsIsNull
+  def rewrite(f: (Expression) => Expression) = f(Or(a.rewriteAsPredicate(f), b.rewriteAsPredicate(f)))
+
+  def arguments = Seq(a, b)
+
+  def symbolTableDependencies = a.symbolTableDependencies ++ b.symbolTableDependencies
+}
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala:43]: new line

Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala:43]: Does this matter? I've seen a bunch of files with this thing on GitHub, but it doesn't seem to have any effect really.

**Comment by petraselmer** [comment]
@boggle and I will be merging manually

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala (hunk 1 above)
Comment (boggle) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala:43]: It is but it also adds the dependencies of predicates.head twice and is harder to read. We've changed it as part of merging

**Comment by petraselmer** [comment]
Manually merged.

File: community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala (hunk 1 above)
Comment (Mats-SX) [inline community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/commands/predicates/Ors.scala:43]: Yeah, sure, you wrote it ;)

**Pull request closed without merging**